### PR TITLE
Polygon: enable handling of holes

### PR DESCRIPTION
**Pull request by eddiemuc**
This PR enables holes in Mapsforge Polygons. Feature was discussed in project c:geo here: https://github.com/cgeo/cgeo/issues/14391

Android Sample was also added in OverlayMapViewer:

image

### CODE REVIEW

File: mapsforge-samples-android/src/main/java/org/mapsforge/samples/android/OverlayMapViewer.java [hunk 1]
```diff
@@ -190,6 +193,52 @@ public boolean onTap(LatLong geoPoint, Point viewPosition,
...
+                addFraction(anchor, 20, 40),
+                addFraction(anchor, 35, 70),
+                addFraction(anchor, 25, 60),
+                addFraction(anchor, 15, 60)));
+
+        return polygonWithHoles;
+    }
+
+    private static LatLong addFraction(final LatLong latLon, final int latAdd, final int lonAdd) {
+        return new LatLong(latLon.getLatitude() + ((double) latAdd / 5000d) , latLon.getLongitude() + ((double) lonAdd / 5000d) );
```
Comment (devemux86) [inline mapsforge-samples-android/src/main/java/org/mapsforge/samples/android/OverlayMapViewer.java:241]: Can you please check the formatting here?

File: mapsforge-map/src/main/java/org/mapsforge/map/layer/overlay/Polygon.java [hunk 2]
```diff
@@ -81,13 +85,44 @@ public synchronized void addPoints(List<LatLong> points) {
...
     }
 
     public synchronized boolean contains(LatLong tapLatLong) {
-        return LatLongUtils.contains(latLongs, tapLatLong);
+        final boolean contains = LatLongUtils.contains(latLongs, tapLatLong);
+        if (holes == null || !contains) {
+            return contains;
+        }
+
+        for(List<LatLong> hole : this.holes) {
```
Comment (devemux86) [inline mapsforge-map/src/main/java/org/mapsforge/map/layer/overlay/Polygon.java:119]: Can you please check the formatting here?

**Comment by devemux86** [comment]
@eddiemuc thanks, very nice work and with example!

If you do not have more changes, we can merge it.
(please check some very minor formatting issues)

File: mapsforge-map/src/main/java/org/mapsforge/map/layer/overlay/Polygon.java (hunk 2 above)
Comment (eddiemuc) [reply mapsforge-map/src/main/java/org/mapsforge/map/layer/overlay/Polygon.java]: I was not sure what formatting you are referring to. I inserted a blank now after the "for". Is this what you meant? (Are there any formatting guidelines for mapsforge code which I missed?)

File: mapsforge-samples-android/src/main/java/org/mapsforge/samples/android/OverlayMapViewer.java (hunk 1 above)
Comment (eddiemuc) [reply mapsforge-samples-android/src/main/java/org/mapsforge/samples/android/OverlayMapViewer.java]: Same here, I was not sure what exactly you referred to. I inserted a line break and removed the blank before the comma and the last parenthesis. If you meant something else I can further modify code

File: mapsforge-map/src/main/java/org/mapsforge/map/layer/overlay/Polygon.java (hunk 2 above)
Comment (devemux86) [reply mapsforge-map/src/main/java/org/mapsforge/map/layer/overlay/Polygon.java]: > I inserted a blank now after the "for".

:+1:

File: mapsforge-samples-android/src/main/java/org/mapsforge/samples/android/OverlayMapViewer.java (hunk 1 above)
Comment (devemux86) [reply mapsforge-samples-android/src/main/java/org/mapsforge/samples/android/OverlayMapViewer.java]: > removed the blank before the comma and the last parenthesis

:+1: 

> I inserted a line break

The new line needs more indentation (not very important).

**Comment by devemux86** [comment]
> Are there any formatting guidelines for mapsforge code which I missed?

@eddiemuc we do not use anything special, just the _default_ IDEA formatting rules.

I run the "Reformat Code" with `Ctrl + Alt + L` and found the issues.
There are some more in imports and comments, but not very important.

File: mapsforge-samples-android/src/main/java/org/mapsforge/samples/android/OverlayMapViewer.java (hunk 1 above)
Comment (eddiemuc) [reply mapsforge-samples-android/src/main/java/org/mapsforge/samples/android/OverlayMapViewer.java]: done

**Comment by eddiemuc** [comment]
Ok I ran the formatting and pushed again

**Comment by devemux86** [comment]
@eddiemuc thanks, looks good to me!

Have you finished, can we merge it?

**Comment by eddiemuc** [comment]
I am finished :-)

**Comment by devemux86** [comment]
@eddiemuc thanks, very nice feature!

**Pull request merged**